cpp/src/strings/convert/convert_durations.cu

```diff
@@ -152,12 +153,8 @@ struct format_compiler {
     }
 
     // create program in device memory
-    d_items.resize(items.size(), stream);
-    CUDF_CUDA_TRY(cudaMemcpyAsync(d_items.data(),
-                                  items.data(),
-                                  items.size() * sizeof(items[0]),
-                                  cudaMemcpyDefault,
-                                  stream.value()));
+    d_items = cudf::detail::make_device_uvector_sync(
+      items, stream, cudf::get_current_device_resource_ref());
```

Comment:
In `convert_datetime.cu`, we create `items` by
```
auto items  = cudf::detail::make_empty_host_vector<format_item>(format.length(), stream);
```

In all cases, I think we can do (a little?) better by storing `items` as member variable and returning without stream sync.

Reply:
I'm not following this.
We don't need the host `items`. It is only created temporarily in host code to copy the vector to device memory in the `d_items` member variable. I don't think this warrants pinned host memory since it is not very big.

Reply:
I meant we can store `items` so it will not be destroyed immediately, then use ` d_items = cudf::detail::make_device_uvector_async` without stream sync. This sounds a bit overkill.

Reply:
Ok, I understand. Yes, I don't want to keep the host items around.